### PR TITLE
MGDOBR-1386: add connector table and DAO for v2

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-1386](https://issues.redhat.com/browse/MGDOBR-1386)

This PR aims to add the `Connectors_V2` table with the following schema

```sql
CREATE TABLE CONNECTOR_V2 (
    id                                         varchar(255) NOT NULL PRIMARY KEY,
    name                                   varchar(255) NOT NULL,
    type                                     varchar(255) NOT NULL, -- The discriminator for sources/sinks
    connector_type_id            varchar(255) NOT NULL,
    connector_external_id     varchar(255),
    topic_name                        varchar(255),
    error                                    text,

    bridge_id                            varchar(255) NOT NULL,

    submitted_at                   timestamp    NOT NULL,
    published_at                   timestamp,

    operation_type                        varchar(255) NOT NULL,
    operation_requested_at         timestamp    NOT NULL,
    operation_completed_at         timestamp,   

    owner                          varchar(255) NOT NULL,

    definition                     jsonb        NOT NULL,

    version                        integer      NOT NULL default 0,
    generation                     integer      NOT NULL default 0,

    unique (bridge_id, name),
    constraint fk_processor foreign key (bridge_id) references BRIDGE_V2 (id)
);

alter table CONDITION add column "connector_id" varchar(255);
alter table CONDITION add foreign key (connector_id) references CONNECTOR_V2(id);
```

and then 2 different DAOs, one for the Sink connectors and one for the Source connectors are added. 
## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-XXX] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/persistence/models/Connector.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                        "left join (" +
+                        "  select connector_id, count(*) as incomplete_count from Condition c where component='MANAGER' and status != 'TRUE' group by c.connector_id " +
+                        "    ) cp on cp.connector_id = conn.id " +
+                        "where " +
+                        // The LEFT JOIN on the sub-query can return a null if there are no MANAGER records that are NOT complete.
+                        "(cp.incomplete_count = 0 or cp.incomplete_count is null) and " +
+                        "b.shard_id = :shardId and " +
+                        "conn.type = :type")
+})
+@FilterDefs({
```
Comment (manstis) [inline manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/persistence/models/Connector.java:55]: You don't have support for filters in your DAO. Did you want to keep the definitions on your Entity?

File: manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/persistence/dao/AbstractConnectorDAOTest.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        Connector connector = Fixtures.createConnector(bridge, getConnectorType());
+        getConnectorDAO().persist(connector);
+
+        Connector retrieved = getConnectorDAO().findByIdWithConditions(connector.getId());
+
+        assertThat(retrieved.getId()).isEqualTo(connector.getId());
+        assertThat(retrieved.getConnectorExternalId()).isEqualTo(TestConstants.DEFAULT_CONNECTOR_EXTERNAL_ID);
+        assertThat(retrieved.getConnectorTypeId()).isEqualTo(TestConstants.DEFAULT_CONNECTOR_TYPE_ID);
+        assertThat(retrieved.getTopicName()).isEqualTo(TestConstants.DEFAULT_CONNECTOR_TOPIC_NAME);
+        assertThat(retrieved.getTopicName()).isEqualTo(TestConstants.DEFAULT_CONNECTOR_TOPIC_NAME);
```
Comment (manstis) [inline manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/persistence/dao/AbstractConnectorDAOTest.java:58]: Duplicate of L#57

File: manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/persistence/dao/AbstractConnectorDAOTest.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        getConnectorDAO().persist(c);
+
+        List<Connector> processors = getConnectorDAO().findByShardIdToDeployOrDelete(b.getShardId());
+        assertThat(processors).isNotNull();
+        assertThat(processors).isEmpty();
+    }
+
+    @Test
+    @Transactional
+    public void testTypeDiscriminator() {
```
Comment (manstis) [inline manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/persistence/dao/AbstractConnectorDAOTest.java:128]: What does this actually test!?!

It creates different types of `Connector`'s and then asserts it cannot be found.

**Review by manstis — COMMENTED** [review]
A few minor comments; but looks pretty good on the whole.

File: manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/persistence/dao/AbstractConnectorDAOTest.java (hunk 3 above)
Comment (r00ta) [reply manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/persistence/dao/AbstractConnectorDAOTest.java:128]: As we use the same table for both sink and sources, it tests that the queries are actually filtering by type. I can remove if you think it is not relevant

Comment (manstis) [reply manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/persistence/dao/AbstractConnectorDAOTest.java:128]: Oh, OK. I see the relevance of the cryptic "type inversion" on L#130.

Create a Connector for the "other type" and it should not be found. Makes sense. Just difficult to see.

Comment (r00ta) [reply manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/persistence/dao/AbstractConnectorDAOTest.java:128]: You're right, I'll add a comment to make it more clear :+1:

File: manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/persistence/models/Connector.java (hunk 1 above)
Comment (r00ta) [reply manager-parent/manager-v2/src/main/java/com/redhat/service/smartevents/manager/v2/persistence/models/Connector.java:55]: I'll remove them, they will be added/tested in the next PRs (https://issues.redhat.com/browse/MGDOBR-1398 for example) :+1:

File: manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/persistence/dao/AbstractConnectorDAOTest.java (hunk 2 above)
Comment (r00ta) [reply manager-parent/manager-v2/src/test/java/com/redhat/service/smartevents/manager/v2/persistence/dao/AbstractConnectorDAOTest.java:58]: wops, thanks :)

**Review by manstis — APPROVED** [review]
LGTM :+1: Thanks for the changes.

**Review by ChughShilpa — APPROVED** [review]

**Pull request merged**